### PR TITLE
Migrate to OAuth2.0

### PR DESCRIPTION
**Pull request by magnific0**
As the title says. I tried to update documentation (README) where I could. I didn't touch the tests, so they still need to be updated. 

I tested the entire OAuth 2.0 process including refreshing the tokens and it works like a charm. Users should be careful to actually use the new ```get_credentials()``` function and update their configuration accordingly. Because access_token, refresh_token, and token_expiry all can change run time.

### CODE REVIEW

File: nokia/__init__.py [hunk 1]
```diff
@@ -223,6 +248,9 @@ def is_measure(self):
 
     def is_target(self):
         return self.category == 2
+
```
Comment (brad) [inline nokia/__init__.py]: Looks like a typo here. Also, this probably doesn't belong in this pull request

Comment (magnific0) [reply nokia/__init__.py]: Yes, you right, I was working with an older file.

**Comment by magnific0** [comment]
I removed the typo you spotted and added the commit to the PR. I'm glad to contribute something back, this library has been very helpful in my projects!

**Comment by wangsha** [comment]
+1
